src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -0,0 +1,124 @@
+    public class ScriptTagHelper : TagHelper
+    {
+        private const string FallbackSrcAttributeName = "asp-fallback-src";
+        private const string FallbackTestMethodAttributeName = "asp-fallback-test";
+        private const string SrcAttributeName = "src";
+
+        // NOTE: All attributes are required for the LinkTagHelper to process.
+        private static readonly string[] RequiredAttributes = new[]
+        {
+            FallbackSrcAttributeName,
+            FallbackTestMethodAttributeName,
+        };
+
+        /// <summary>
+        /// The URL of a Script tag to fallback to in the case the primary one fails (as specified in the src
+        /// attribute).
+        /// </summary>
+        [HtmlAttributeName(FallbackSrcAttributeName)]
+        public string FallbackSrc { get; set; }
+
+        /// <summary>
+        /// The script method defined in the primary script to use for the fallback test.
+        /// </summary>
+        [HtmlAttributeName(FallbackTestMethodAttributeName)]
+        public string FallbackTestMethod { get; set; }
+
+        // Protected to ensure subclasses are correctly activated. Internal for ease of use when testing.
+        [Activate]
+        protected internal ILogger<ScriptTagHelper> Logger { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            // TODO: Consider using the RazorStringWriter instead of a StringBuilder, because Script tags may contain large bodies.
+
+            if (!context.AllRequiredAttributesArePresent(RequiredAttributes, Logger))
+            {
+                if (Logger.IsEnabled(LogLevel.Verbose))
+                {
+                    Logger.WriteVerbose("Skipping processing for {0} {1}", nameof(ScriptTagHelper), context.UniqueId);
+                }
+
+                return;
+            }
+
+            var content = new StringBuilder();
+
+            // NOTE: Values in TagHelperOutput.Attributes are already HtmlEncoded
+
+            // We've taken over rendering here so prevent the element rendering the outer tag
+            output.TagName = null;
+
+            // Rebuild the <script /> tag.
+            content.Append("<script");
+            foreach (var attribute in output.Attributes)
+            {
+                content.AppendFormat(CultureInfo.InvariantCulture, " {0}=\"{1}\"", attribute.Key, attribute.Value);
+            }
+
+            content.Append(">");
+
+            var originalContent = await context.GetChildContentAsync();
+            content.Append(originalContent);
+            content.AppendLine("</script>");
+
+            // Build the <script /> tag that checks the test method and if it fails, renders the extra script.
+            content.Append("<script>(");
+            content.Append(FallbackTestMethod);
+            content.Append(" || document.write(\"<script");
+
+            if (!output.Attributes.ContainsKey("src"))
```

Comment:
Now since the old diff is outdated, re-surfacing previous comment:
> Also, just create a new attributes dictionary here and add a src attribute with the fallback src url. That way your foreach below can just append attributes without special casing the src attribute name. By doing this you can also inline the AppendSrc method here.